src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess.swift

```diff
@@ -74,20 +74,38 @@ struct CertificateAccess {
             return .failure(.HC_CBORWEBTOKEN_NO_DIGITALGREENCERTIFICATE)
         }
 
-        let _cborData = healthCertificateCBOR.encode()
-        let cborData = Data(_cborData)
+        switch validateSchema(of: healthCertificateCBOR) {
+        case .success:
+            let _cborData = healthCertificateCBOR.encode()
+            let cborData = Data(_cborData)
+            let codableDecoder = CodableCBORDecoder()
 
-        let codableDecoder = CodableCBORDecoder()
+            guard let healthCertificate = try? codableDecoder.decode(DigitalGreenCertificate.self, from: cborData) else {
+                return .failure(.HC_CBOR_DECODING_FAILED)
+            }
+            return .success(healthCertificate)
 
-        guard let healthCertificate = try? codableDecoder.decode(DigitalGreenCertificate.self, from: cborData) else {
-            return .failure(.HC_CBOR_DECODING_FAILED)
+        case let .failure(error):
+            return .failure(error)
         }
-
-        return .success(healthCertificate)
     }
 
     // MARK: - Private
 
+    private func validateSchema(of certificate: CBOR) -> Result<Void, CertificateDecodingError> {
+        guard case let CBOR.map(certificateMap) = certificate,
+              let schemaURL = Bundle.module.url(forResource: "CertificateSchema", withExtension: "json"),
+              let schemaData = FileManager.default.contents(atPath: schemaURL.path),
+              let schemaDict = try? JSONSerialization.jsonObject(with: schemaData) as? [String: Any],
+              let validationResult = try? JSONSchema.validate(certificateMap, schema: schemaDict),
+              case .valid = validationResult else {
+
```

Comment:
perhaps add a log here?

Reply:
Logging is not possible at the moment. We need to discuss a solution in the DEV-Camp. I have added an entry for that.